tests/filecheck/transforms/scf_for_loop_range_folding.mlir

```diff
@@ -0,0 +1,95 @@
+// RUN: xdsl-opt -p scf-for-loop-range-folding --split-input-file %s | filecheck %s
+
+func.func @fold_one_loop(%arg0: memref<?xi32>, %arg1: index, %arg2: index) {
```

Comment:
what is the purpose of func op args? maybe could be result of test ops inline also?

Reply:
Yeah, definitely. I just decided to copy tests from mlir. I think they use function arguments to avoid definitions through test.op everywhere. Do you think it's worth changing to test.op?

Reply:
Probably, and also renaming the values to be a bit more readable, like `%lb`, `%ub` and `%step` for clarity. We definitely don't need the memref though